lib/lp_token_auth/config.rb

```diff
@@ -9,22 +9,29 @@ class Config
     # * `expires` is an integer corresponding to the number of hours that the token is active
     # * `secret` is a string corresponding to the secret key used when encrypting the token
     # * `token_transport` is a string indicating where to include the token in the HTTP response
-    attr_accessor :algorithm, :expires, :secret, :token_transport
+    attr_accessor :algorithm, :expires, :secret, :token_transport, :jwe_private_key, :jwe_encryption
 
     # Provides default values to token options
+    # ENV defaults defined as procs to ensure they return their
+    # latest value at call time (else they return nil,
+    # since ENV values may not be initialized before gem code)
     DEFAULT_VALUES = {
       algorithm: 'HS512',
       expires: (7 * 24),
       token_transport: [:cookie],
+      jwe_private_key: -> { ENV['JWE_PRIVATE_KEY'] },
+      jwe_encryption: -> { ENV['JWE_ENCRYPTION'] || 'A256GCM' }
     }
 
     # Retrieves value for token option, either as set by the application, or the default
     # @param [Symbol] key the token option name
     # @raise [LpTokenAuth::Error] if the option has not been set by the application and a default value does not exist
     # @return [String,Integer] the value of the token option
     def get_option(key)
-      option = send(key) || DEFAULT_VALUES[key]
-      raise LpTokenAuth::Error "Missing config option value: #{ key }" unless option
+      default = DEFAULT_VALUES[key]
+      default = default.call if default.is_a?(Proc)
```

Comment:
N/B" could probably extract this line and the one above into a private `default(key)` method